syntax = "proto2";

import "lib/common.proto";

message Tids990DecoderProto {}

message Tids990EncoderProto {
	optional double rotational_period_ms = 1 [ default = 166,
		(help) = "length of a track" ];
	optional int32 sector_count = 2 [ default = 26,
		(help) = "number of sectors per track" ];
	optional double clock_period_us = 3 [ default = 2,
		(help) = "clock rate of data to write" ];
	optional int32 am1_byte = 4 [ default = 0x2244,
		(help) = "16-bit RAW bit pattern to use for the AM1 ID byte" ];
	optional int32 am2_byte = 5 [ default = 0x2245,
		(help) = "16-bit RAW bit pattern to use for the AM2 ID byte" ];
	optional int32 gap1_bytes = 6 [ default = 80,
		(help) = "size of gap 1 (the post-index gap)" ];
	optional int32 gap2_bytes = 7 [ default = 21,
		(help) = "size of gap 2 (the post-ID gap)" ];
	optional int32 gap3_bytes = 8 [ default = 51,
		(help) = "size of gap 3 (the post-data or format gap)" ];
}

